server/game/gameSystems/DiscardFromDeckSystem.ts

```diff
@@ -36,21 +25,27 @@ export class DiscardFromDeckSystem<TContext extends AbilityContext = AbilityCont
     public override canAffect(player: Player, context: TContext, additionalProperties = {}, mustChangeGameState = GameStateChangeRequired.None): boolean {
         const properties = this.generatePropertiesFromContext(context, additionalProperties);
 
-        const availableDeck = player.drawDeck || []; // Without this | [], sometimes this is undefined, causing issues below
+        const availableDeck = Array.isArray(player) ? player[0].drawDeck : player.drawDeck;
```

Comment:
Ah yeah this needs to be a for loop then, if multiple player targets are provided it needs to act on all of them. Same as it's done for DiscardCardsFromHand